core/shared/src/main/scala/zio/ZSchedule.scala

```diff
@@ -248,15 +243,29 @@ trait ZSchedule[-R, -A, +B] extends Serializable { self =>
     new ZSchedule[R, A1, B] {
       type State = self.State
       val initial = self.initial
-      val update  = (a: A1, s: State) => self.update(f(a), s)
+      val extract = (a: A1, s: self.State) => self.extract(f(a), s)
+      val update  = (a: A1, s: self.State) => self.update(f(a), s)
     }
 
   /**
-   * Returns a new schedule with the specified pure modification
-   * applied to each delay produced by this schedule.
+   * Returns a new schedule with the given delay added to every update.
+   */
+  final def delayed(f: B => Duration): ZSchedule[R with Clock, A, B] =
```

Comment:
Hmm, maybe call this `addDelay`, and preserve the old `delayed`? In theory we can still have the old one for `R = ZEnv`.

Reply:
👍 Will do